build(docs-infra): update @angular/* to 12.1.1

gkalpak opened the pull request:
Update `@angular/*` packages to latest 12.1.x versions (mainly to take advantage of recent ServiceWorker improvements, such as #42607 and #42622).

mary-poppins commented:
You can preview 26d915d at https://pr42776-26d915d.ngbuilds.io/.

gkalpak commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -2,10 +2,10 @@
-        "polyfills-es2017": 55210,
-        "styles": 69211,
+        "runtime-es2017": 4841,
+        "main-es2017": 458609,
```
FYI, from the ~2KB increase, ~0.5KB come from updating CLI/devkit and ~1.5KB from updating CDK.

mary-poppins commented:
You can preview aa33ec5 at https://pr42776-aa33ec5.ngbuilds.io/.

mary-poppins commented:
You can preview 1f18575 at https://pr42776-1f18575.ngbuilds.io/.

petebacondarwin commented on aio/src/app/documents/document.service.spec.ts:
```diff
@@ -77,6 +77,7 @@ describe('DocumentService', () => {
       httpMock.expectOne(CONTENT_URL_PREFIX + 'doc__with__underscores.json').flush({});
       locationService.go('DOC_WITH_UNDERSCORES');
       httpMock.expectOne(CONTENT_URL_PREFIX + 'd_o_c___w_i_t_h___u_n_d_e_r_s_c_o_r_e_s_.json').flush({});
+      expect().nothing();
```
This makes me a bit sad...

petebacondarwin replied:
Another way to get around this would be to wrap the test code in a `expect(() => {...}).not.toThrow()`?

gkalpak replied:
Switched to `.not.toThrow()`. (Making you sad is the last thing I want from my PRs :grin:)

petebacondarwin reviewed (approved):
Typo in first commit: `en eminent` -> `an imminent`

petebacondarwin reviewed (approved):
Reviewed-for: size-tracking

gkalpak committed:
build(docs-infra): update payload sizes

This commit updates the payload sizes for angular.io to make it easier
to identify payload size changes from an imminent update of `@angular/*`
packages.

gkalpak committed:
build(docs-infra): update @angular/* to 12.1.1

Update `@angular/*` packages to latest 12.1.x versions (mainly to take
advantage of recent ServiceWorker improvements, such as #42607
and #42622).

gkalpak committed:
test(docs-infra): prevent warning due to missing expectation

Fix a unit test warning due to Jasmine not realizing that
`httpMock.expectOne()` is an expectation.

[Example][1]:

> WARN: 'Spec 'DocumentService currentDocument should encode the request
> path to be case-insensitive' has no expectations.'

[1]: https://circleci.com/gh/angular/angular/1017640

gkalpak committed:
refactor(docs-infra): replace deprecated Sass `/` division with `math.div()`

This commit replaces the deprecated `/` operators used for division in
Sass files with the [recommended][1] `math.div()` function to get rid of
build warnings ([example][2]).

[1]: https://sass-lang.com/documentation/breaking-changes/slash-div
[2]: https://circleci.com/gh/angular/angular/1017640

gkalpak committed:
fixup! test(docs-infra): prevent warning due to missing expectation

gkalpak committed:
fix(docs-infra): log more SW debug info

This commit expands the info printed to the console to help diagnose
ServiceWorker issues to include the [internal debug info][1] retrieved
from `/ngsw/state`. This will provide more useful data, such as the
activated SW's version, state, clients, recent operations and any recent
errors.

NOTE:
This temporarily increases the payload size. Removing this code and
reclaiming the payload size is being tracked in #41117.

[1]: https://angular.io/guide/service-worker-devops#locating-and-analyzing-debugging-information

mary-poppins commented:
You can preview 52c54ce at https://pr42776-52c54ce.ngbuilds.io/.

petebacondarwin reviewed (approved):
Reviewed-for: size-tracking

atscott reviewed (approved):
Reviewed-for: size-tracking
